tests/v2/integration/catalogv2/ui_plugin_test.go

```diff
@@ -0,0 +1,369 @@
+	corev1           corev1.CoreV1Interface
+	originalBranch   string
+	originalGitRepo  string
+}
+
+func (w *UIPluginTest) TearDownSuite() {
+	w.Require().NoError(w.uninstallApp(namespace.UIPluginNamespace, "uk-locale"))
+	w.Require().NoError(w.uninstallApp(namespace.UIPluginNamespace, "clock"))
+	w.Require().NoError(w.catalogClient.ClusterRepos().Delete(context.Background(), "extensions-examples", metav1.DeleteOptions{PropagationPolicy: &propagation}))
+	w.session.Cleanup()
+
+}
+
+func (w *UIPluginTest) SetupSuite() {
+	var err error
+	testSession := session.NewSession()
+	w.session = testSession
+	w.client, err = rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+	insecure := true
+	w.client.RancherConfig.Insecure = &insecure
+	w.catalogClient, err = w.client.GetClusterCatalogClient("local")
+	require.NoError(w.T(), err)
+
+	kubeConfig, err := kubeconfig.GetKubeconfig(w.client, "local")
+	require.NoError(w.T(), err)
+
+	restConfig, err := (*kubeConfig).ClientConfig()
+	require.NoError(w.T(), err)
+
+	//restConfig.Insecure = true
+
+	cset, err := kubernetes.NewForConfig(restConfig)
+	require.NoError(w.T(), err)
+	w.corev1 = cset.CoreV1()
+
+	w.restClientGetter, err = kubeconfig.NewRestGetter(restConfig, *kubeConfig)
+	require.NoError(w.T(), err)
+	require.NoError(w.T(), err)
+	_, err = w.catalogClient.ClusterRepos().Create(context.Background(), &rv1.ClusterRepo{
+		ObjectMeta: metav1.ObjectMeta{Name: "extensions-examples"},
+		Spec:       rv1.RepoSpec{GitRepo: "https://github.com/diogoasouza/ui-plugin-examples", GitBranch: "main"},
+	}, metav1.CreateOptions{})
+	w.Require().NoError(err)
+	w.Require().NoError(w.pollUntilDownloaded("extensions-examples", metav1.Time{}))
+	w.Require().NoError(w.catalogClient.InstallChartFromRepo(&types.ChartInstallAction{
+		DisableHooks:             false,
+		Timeout:                  &metav1.Duration{Duration: 60 * time.Second},
+		Wait:                     true,
+		Namespace:                namespace.UIPluginNamespace,
+		DisableOpenAPIValidation: false,
+		Charts: []types.ChartInstall{{
+			ChartName:   "uk-locale",
+			Version:     "0.1.0",
+			ReleaseName: "uk-locale",
+			Description: "locale",
+		}},
+	}, "extensions-examples"))
+	w.Require().NoError(w.waitForChart(rv1.StatusDeployed, "uk-locale", 0))
+
+	w.Require().NoError(w.catalogClient.InstallChartFromRepo(&types.ChartInstallAction{
+		DisableHooks:             false,
+		Timeout:                  &metav1.Duration{Duration: 60 * time.Second},
+		Wait:                     true,
+		Namespace:                namespace.UIPluginNamespace,
+		DisableOpenAPIValidation: false,
+		Charts: []types.ChartInstall{{
+			ChartName:   "clock",
+			Version:     "0.2.0",
+			ReleaseName: "clock",
+			Description: "clock",
+		}},
+	}, "extensions-examples"))
+	w.Require().NoError(w.waitForChart(rv1.StatusDeployed, "clock", 0))
+
+	//Waiting for controller cache to update
+	time.Sleep(10 * time.Second)
+}
+
+func (w *UIPluginTest) resetSettings() {
```

Comment:
this func is empty

Reply:
I was keeping it just to follow the pattern, but I removed it